mil_common/simulation/mil_gazebo/src/mil_inclinometer_gazebo.cpp

```diff
@@ -0,0 +1,69 @@
+#include <mil_gazebo/mil_inclinometer_gazebo.hpp>
+
+namespace mil_gazebo
+{
+// Register this plugin
+GZ_REGISTER_SENSOR_PLUGIN(MilInclinometerGazebo)
+
+MilInclinometerGazebo::MilInclinometerGazebo() : SensorPlugin()
+{
+}
+
+MilInclinometerGazebo::~MilInclinometerGazebo()
+{
+}
+
+void MilInclinometerGazebo::Load(gazebo::sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
+{
+  sensor_ = std::dynamic_pointer_cast<gazebo::sensors::ImuSensor>(_parent);
+  if (!sensor_)
+  {
+    ROS_ERROR_NAMED("MILInclinometerGazebo", "Parent sensor is null");
+    return;
+  }
+
+  parent_ = boost::dynamic_pointer_cast<gazebo::physics::Link>(
+      gazebo::physics::get_world(sensor_->WorldName())->EntityByName(sensor_->ParentName()));
+  if (!parent_)
+  {
+    ROS_ERROR_NAMED("MILInclinometerGazebo", "Parent link is null");
+    return;
+  }
+
+  if (_sdf->HasElement("frame_id"))
+  {
+    frame_name = _sdf->GetElement("frame_id")->Get<std::string>();
+  }
+  else
+  {
+    frame_name = sensor_->ParentName();
+  }
+
+  msg_.incremental_velocity_covariance[0] =
+      NoiseCovariance(*sensor_->Noise(gazebo::sensors::SensorNoiseType::IMU_LINACC_X_NOISE_METERS_PER_S_SQR));
+  msg_.incremental_velocity_covariance[4] =
+      NoiseCovariance(*sensor_->Noise(gazebo::sensors::SensorNoiseType::IMU_LINACC_Y_NOISE_METERS_PER_S_SQR));
+  msg_.incremental_velocity_covariance[8] =
+      NoiseCovariance(*sensor_->Noise(gazebo::sensors::SensorNoiseType::IMU_LINACC_Z_NOISE_METERS_PER_S_SQR));
+
+  pub_ = nh_.advertise<mil_msgs::IncrementalVelocity>("/imu/inclinometer", 20);
+
+  update_connection_ = sensor_->ConnectUpdated(boost::bind(&MilInclinometerGazebo::OnUpdate, this));
+}
+
+void MilInclinometerGazebo::OnUpdate()
```

Comment:
Make sure that this will only report incremental velocity at time intervals possible by the physical IMU, and make that amount configurable in the xacro (so that if we switch to another IMU / reconfigure this one to run at a diffrent clock speed, the data still makes sense).

Reply:
I've changed it so the plugin reads in update_rate from the xacro and plugs it in to ros::rate loop_rate(update_rate). I've noticed that other plugins do not do this, is this the proper way of doing this?

Reply:
ya so typically, gazebo plugin use gazebo time not ros time. Here is an example of another plugin doing this. https://github.com/uf-mil/vrx/blob/8c204f359effb8ab810fac6b57ebc46d13dcaf3a/vrx_gazebo/src/light_buoy_plugin.cc#L227